### PR TITLE
Remove 'auto refresh' feature

### PR DESCRIPTION
**Pull request by daniel-beck**
This is a re-done https://github.com/jenkinsci/jenkins/pull/2979 after https://github.com/jenkinsci/jenkins/pull/3894.

#### Telemetry

Looking at the telemetry submitted in November and December 2019, which may include _some_ 2.204.1 LTS, but otherwise mostly weeklies:

67880 instances reported data. About 10% of instances reported _any_ use of auto-refresh during these two months (i.e. it wasn't used _at all_ on ~90% of instances).

Auto-refresh was disabled in 166555708 (99.4%) sessions, and enabled in 969858 sessions (0.6%).

This result is probably skewed by auto refresh keeping sessions alive on more pages, but even so, what's the real ratio? Maybe 2%?

I'd be happy to make this data available if desired, but I'm OK with the result (and previous reviewers in #2979 didn't care at all).

### Proposed changelog entries

* Remove 'auto refresh' feature, including now obsolete auto refresh telemetry capability.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

File: core/src/main/resources/lib/layout/html.jelly [hunk 1]
```diff
@@ -37,8 +37,7 @@ THE SOFTWARE.
       Title of the HTML page. Rendered into &lt;title> tag.
     </st:attribute>
     <st:attribute name="norefresh">
-      If non-null and not "false", auto refresh is disabled on this page.
-      This is necessary for pages that include forms.
+      Deprecated: Used to disable auto refresh for a page. The feature has been removed.
```
Comment (timja) [inline core/src/main/resources/lib/layout/html.jelly]:
```suggestion
      Deprecated: Was used to disable auto refresh for a page. The feature has been removed.
```

**Review by timja — APPROVED** [review]
I've done a search around, only possible reference I could find was https://github.com/jenkinsci/jenkins/blob/master/war/src/main/webapp/scripts/yui/treeview/treeview-debug.js#L1109, but I can't tell if its related tbh...

File: core/src/main/java/hudson/Functions.java [hunk 2]
```diff
@@ -633,54 +632,17 @@ public static String getYuiSuffix() {
...
     /**
-     * Used by {@code layout.jelly} to control the auto refresh behavior.
+     * No longer used.
      *
-     * @param noAutoRefresh
-     *      On certain pages, like a page with forms, will have annoying interference
-     *      with auto refresh. On those pages, disable auto-refresh.
+     * @deprecated auto refresh has been removed
      */
+    @Deprecated
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/Functions.java]: Also restrict it to force migration?

**Review by oleg-nenashev — APPROVED** [review]
+1 for removal. Would be great to get it into the next LTS.

P.S: with all recent changes, time to introduce the "removed" category in Jenkins changelogs

**Review by markjacksonfishing — APPROVED** [review]

**Comment by batmat** [comment]
FYI @fqueiruga @alecharp @varyvol

**Comment by daniel-beck** [comment]
>  @oleg-nenashev added the removed label 2 hours ago

Please add a description for the label. It's really unclear what its purpose is. How is this different from `major-rfe`? Why is this PR labeled both?

**Comment by oleg-nenashev** [comment]
> Please add a description for the label. It's really unclear what its purpose is. How is this different from major-rfe? Why is this PR labeled both?

Added description. The PR is labeled with both, because we do not have support of `removed` on the Changelog generator side or in the changelog renderer. If there is a consensus we want to have one, I will do updates accordingly

**Comment by fqueiruga** [comment]
I'll monitor this PR and will update the header PR as soon as it's merged.

**Review by Wadeck — APPROVED** [review]
Thanks for the determination on this long story :) I hope this PR will be the last one 🎉 

The telemetry was really useful to discover/understand the behavior 💯 

Also, the PR is fairly long but the change are minimum, that eases the review process!

**Comment by oleg-nenashev** [comment]
I suggest merging it tomorrow if there is no negative feedback

**Comment by oleg-nenashev** [comment]
Would it make sense to add some upgrade guidelines here for the users who still use this feature?

**Comment by daniel-beck** [comment]
What would it say? It's not like they can set a system property and re-enable it.

Anyway, this isn't an important change, so I'm on-holding it to reduce risk to the upcoming weekly not being a good LTS baseline.

**Review by alecharp — APPROVED** [review]

**Review by fcojfernandez — APPROVED** [review]
+1 to this change. I'm approving but adding the ` unresolved-merge-conflict` label

**Comment by daniel-beck** [comment]
I have now had to resolve merge conflicts three times this week.

Un-on-holding this, is everyone else wants their badly scheduled last minute changes in 2.222 and does their best to prevent it from being a reasonable LTS baseline, I want this in, too.

**Review by oleg-nenashev — APPROVED** [review]
Let's ship it. 24hrs interval has already passed, so no need to wait

**Comment by amdoucette** [comment]
Just saw this on the 2.223 release notes. Disappointed as we use auto-refresh (we've got a TV showing a specific page, auto-refreshing so the dev/QA group can easily see build status without having to filter through many emails or open the page locally). I'm sure we'll find a workaround, but we did use and enjoy this functionality.

**Comment by daniel-beck** [comment]
@amdoucette 

> we've got a TV showing a specific page

Install any one of a hundred browser plugins that can reload a page being shown periodically.

**Pull request merged**